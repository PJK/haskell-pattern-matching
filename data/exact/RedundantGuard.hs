module RedundantGuard where


func :: Int -> Int
func x
    | x == 0 = 0
    | x > 0 = 1
    | x < 0 = -1
    | x == 3 = 3