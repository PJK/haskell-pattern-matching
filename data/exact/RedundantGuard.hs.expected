{
    "tag": "AnalysisSuccess",
    "contents": [
        [
            "func",
            {
                "tag": "Redundant",
                "contents": [
                    {
                        "tag": "WildcardPattern",
                        "contents": []
                    },
                    {
                        "tag": "GuardPattern",
                        "contents": [
                            {
                                "tag": "ConstructorPattern",
                                "contents": [
                                    "True",
                                    []
                                ]
                            },
                            {
                                "tag": "BExp",
                                "contents": {
                                    "tag": "IntBoolOp",
                                    "contents": [
                                        "IntEQ",
                                        {
                                            "tag": "IntVar",
                                            "contents": "x"
                                        },
                                        {
                                            "tag": "IntLit",
                                            "contents": 3
                                        }
                                    ]
                                }
                            }
                        ]
                    }
                ]
            }
        ]
    ]
}
